fsnotify_test.go

```diff
@@ -892,6 +892,28 @@ func TestClose(t *testing.T) {
 
 		chanClosed(t, w.w)
 	})
+
+	t.Run("error after closed", func(t *testing.T) {
+		t.Parallel()
+
+		tmp := t.TempDir()
+		w := newWatcher(t, tmp)
+		if err := w.Close(); err != nil {
+			t.Fatal(err)
+		}
+
+		file := filepath.Join(tmp, "file")
+		touch(t, file)
+		if err := w.Add(file); !errors.Is(err, ErrClosed) {
+			t.Fatalf("wrong error for Add: %#v", err)
+		}
+		if err := w.Remove(file); err != nil {
+			t.Fatalf("wrong error for Remove: %#v", err)
+		}
+		if ugh := w.WatchList(); ugh != nil { // This function is crap.
+			t.Fatalf("WatchList not nil: %#v", ugh)
```

Comment:
Probably better to use err instead of ugh. That said, I know the feeling!

Reply:
Ah, I get it now. Thanks for the improvement in f8aebb4!

Reply:
I just get annoyed every time I see this function because everything about it is wrong IMO, which is impressive for such a small feature 🙃